macros/is_databricks_sql_warehouse.sql

```diff
@@ -0,0 +1,15 @@
+{% macro is_databricks_sql_warehouse(target) %}
+    {% if target.type in ('databricks','spark') %}
+        {% set re = modules.re %}
+        {% set path_match = target.http_path %}
+        {% set regex_pattern = "/sql/.+/warehouses/" %}
+        {% set match_result = re.search(regex_pattern, path_match) %}
+        {% if match_result %}
+            {{ return(True) }}
+        {% else %}
+            {{ return(False) }}
+        {% endif %}
+    {% else %}
+        {{ return(False) }}
+    {% endif %}
+{% endmacro %}
```

Comment:
I also have an [open question on dbt Slack](https://getdbt.slack.com/archives/CNGCW8HKL/p1712069128469359) which asks if there is a better way to do this natively using the dbt-databricks adapter.

Reply:
It seems per dbt Slack that this is the preferred route.